shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -215,30 +216,54 @@ private void setSystemChromeApplicationSwitcherDescription(
     }
   }
 
-  private void setSystemChromeChangeListener() {
-    // Set up a listener to notify the framework when the system ui has changed.
+  private void setSystemChromeChangeListenerLegacy() {
     View decorView = activity.getWindow().getDecorView();
     decorView.setOnSystemUiVisibilityChangeListener(
         new View.OnSystemUiVisibilityChangeListener() {
           @Override
           public void onSystemUiVisibilityChange(int visibility) {
             if ((visibility & View.SYSTEM_UI_FLAG_FULLSCREEN) == 0) {
-              // The system bars are visible. Make any desired adjustments to
-              // your UI, such as showing the action bar or other navigational
-              // controls. Another common action is to set a timer to dismiss
-              // the system bars and restore the fullscreen mode that was
-              // previously enabled.
               platformChannel.systemChromeChanged(false);
             } else {
-              // The system bars are NOT visible. Make any desired adjustments
-              // to your UI, such as hiding the action bar or other
-              // navigational controls.
               platformChannel.systemChromeChanged(true);
             }
           }
         });
   }
 
+  private void setSystemChromeChangeListener() {
+    // Set up a listener to notify the framework when the system ui has changed.
+    // The Android API for overlays/insets as of API 30 provides backwrds compatibility
```

Comment:
```suggestion
    // The Android API for overlays/insets as of API 30 provides backwards compatibility
```

Reply:
@Piinks I tried this but setting the visibility of the system bars with the mock was not getting picked up by the listener :/ Do you have any examples of testing a listener like this? The one I referred to earlier was [this](https://github.com/flutter/engine/blob/c058dd1896010a03f2dc60056d52b66de0faab26/shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java#L2030).